File: Source/Fantasy.Jobs/TrackProgressService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Fantasy.ServiceModel;
using Fantasy.Tracking;

namespace Fantasy.Jobs
{
    public class TrackProgressService : AbstractService,  IProgressMonitor
    {

        private ITrackProvider _track;

        public override void InitializeService()
        {
            _track = this.Site.GetService<ITrackProvider>();
            base.InitializeService();
        }

        #region IProgressMonitor Members

        public int Value
        {
            get
            {
                return _track.GetProperty("progress.value", 0); 
            }
            set
            {
                _track["progress.value"] = value; 
            }
        }

        public int Maximum
        {
            get
            {
                return _track.GetProperty("progress.maximum", 100);
            }
            set
            {
                _track["progress.maximum"] = value;
            }
        }

        public int Minimum
        {
            get
            {
                return _track.GetProperty("progress.minimum", 0);
            }
            set
            {
                _track["progress.minimum"] = value;
            }
        }

        #endregion

        #region IProgressMonitor Members


        public ProgressMonitorStyle Style
        {
            get
            {
                return _track.GetProperty("progress.style", ProgressMonitorStyle.Blocks);
            }
            set
            {
                _track["progress.style"] = value;
            }
        }

        #endregion
    }
}

File: branches/Stheno/AutoConverted/Fantasy.Jobs/TrackProgressService.java
package Fantasy.Jobs;

import Fantasy.ServiceModel.*;
import Fantasy.Tracking.*;

public class TrackProgressService extends AbstractService implements IProgressMonitor
{

	private ITrackProvider _track;

	@Override
	public void InitializeService()
	{
		_track = this.Site.<ITrackProvider>GetService();
		super.InitializeService();
	}

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#region IProgressMonitor Members

	public final int getValue()
	{
		return _track.GetProperty("progress.value", 0);
	}
	public final void setValue(int value)
	{
		_track["progress.value"] = value;
	}

	public final int getMaximum()
	{
		return _track.GetProperty("progress.maximum", 100);
	}
	public final void setMaximum(int value)
	{
		_track["progress.maximum"] = value;
	}

	public final int getMinimum()
	{
		return _track.GetProperty("progress.minimum", 0);
	}
	public final void setMinimum(int value)
	{
		_track["progress.minimum"] = value;
	}

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#endregion

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#region IProgressMonitor Members


	public final ProgressMonitorStyle getStyle()
	{
		return _track.GetProperty("progress.style", ProgressMonitorStyle.Blocks);
	}
	public final void setStyle(ProgressMonitorStyle value)
	{
		_track["progress.style"] = value;
	}

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#endregion
}